lib/SDIR/Dialect/Ops.cpp

```diff
@@ -850,6 +868,9 @@ std::string AllocOp::getName() {
   std::string name;
   llvm::raw_string_ostream nameStream(name);
   (*this)->getResult(0).printAsOperand(nameStream, state);
+  // replace %-sign with underscore
+  name.erase(0, 1);
```

Comment:
You're probably going to need a more general utility function (e.g., also dashes (-) have to be replaced with underscores), I propose `sanitizeName` but your call